### PR TITLE
Fix Quaternions.derivate method and :

### PR DESCRIPTION
**Pull request by fbailly**
Replaced eigen::Vector with biorbd::utils::Vector,
Added possibility to specify Kstabilizer in Quaternion constructor,
Thus changed the constructors and modified Joints.cpp and Vector.cpp accordingly
Added method get_array in python binding

### CODE REVIEW

File: src/Utils/Quaternion.cpp [hunk 1]
```diff
@@ -3,37 +3,41 @@
 
 #include <Eigen/Dense>
```
Comment (pariterre) [inline src/Utils/Quaternion.cpp]: Is that possible to add a test for the derivative function. I am concern by the changes in the order of the constructor where we swapped things around!
Thanks :D

File: test/test_rigidbody.cpp [hunk 2]
```diff
@@ -269,6 +269,18 @@ TEST(Dynamics, ForwardAccelerationConstraint){
         EXPECT_NEAR(cs.force[i], forces_expected[i], requiredPrecision);
 }
 
+TEST(Kinematics, computeQdot)
+{
+    biorbd::Model m("models/simple_quat.bioMod");
+    biorbd::rigidbody::GeneralizedCoordinates Q_quat(m.nbQ()), QDot(m), QDot_quat(m.nbQ()), QDot_quat_expected(m.nbQ());
+    Q_quat << 0,0,0,1;
```
Comment (pariterre) [inline test/test_rigidbody.cpp]: Is there a risk of false positive in using the unitary quaternion?

File: test/test_utils.cpp [hunk 3]
```diff
@@ -386,3 +387,67 @@ TEST(ModelReading, equations)
         EXPECT_DOUBLE_EQ(node.x(), node.y());
 }
 
+TEST(Quaternion, addition)
+{
+    biorbd::utils::Quaternion q1(1,2,3,4);
+    biorbd::utils::Quaternion q2(2,3,4,5);
+    EXPECT_NEAR(q1[3], 1, requiredPrecision);
```
Comment (pariterre) [inline test/test_utils.cpp]: Hum... Does it feels like a bug to you (In a user perspective I mean) that the components shuffle around? If you think so, I can take it from there to make it consistent

File: test/test_utils.cpp [hunk 4]
```diff
@@ -386,3 +387,67 @@ TEST(ModelReading, equations)
...
+    biorbd::utils::Quaternion q12 = q1*q2;
+    EXPECT_NEAR(q12[3], -36, requiredPrecision);
+    EXPECT_NEAR(q12[0], 6, requiredPrecision);
+    EXPECT_NEAR(q12[1], 12, requiredPrecision);
+    EXPECT_NEAR(q12[2], 12, requiredPrecision);
+}
+
+TEST(Quaternion, derivative)
+{
+    biorbd::utils::Quaternion q1;
```
Comment (pariterre) [inline test/test_utils.cpp]: Again, is there a risk of false positive when using the unitary quaternion?

File: test/test_utils.cpp (hunk 3 above)
Comment (fbailly) [reply test/test_utils.cpp]: It is a matter of convention. In rbdl they are defined as (x,y,z,w). But in the biorbd constructors, we use multiple logics. I.E : (w,x,y,z), or directly a vector which will be interpreted in rbdl as (x,y,w,z). Why not using a single constructor which would call the rbdl default one (using whatever convention, I personally prefer (w,x,y,z)), and then all the other constructors should refer to this one, in order to limit the possible confusions.
The reason why the test is performed in this order is arbitrary though.

File: test/test_rigidbody.cpp (hunk 2 above)
Comment (fbailly) [reply test/test_rigidbody.cpp]: I don't think so, but I'll change that in order to be 100% sure

File: test/test_utils.cpp (hunk 4 above)
Comment (fbailly) [reply test/test_utils.cpp]: Idem

**Comment by pariterre** [comment]
Is that PR still in [WIP]?

**Comment by fbailly** [comment]
There is still some work to do across biorbd on the inputs and outputs of fonctions to harmonize the use of biorbd objects instead of Eigen ones (for a smoother use in python).
However, this might be the subject of another PR. I suggest that we make sure that the Quaternion class is clean wrt this matter before closing the present PR.

**Comment by fbailly** [comment]
There is still some work to do across biorbd on the inputs and outputs of fonctions to harmonize the use of biorbd objects instead of Eigen ones (for a smoother use in python).
However, this might be the subject of another PR. I suggest that we make sure that the Quaternion class is clean wrt this matter before closing the present PR.

**Comment by pariterre** [comment]
I agree,
In this regard, do you think Matrix3d should be interfaced as well so it can be used in Python or the RotoTrans is enough?

**Comment by fbailly** [comment]
Yes why not. Otherwise, methods such as Quaternion.fromMatrix() or some of the constructors of RotoTrans() cannot be be used in python.

**Pull request merged**